lib/ansible/module_utils/ovirt.py

```diff
@@ -0,0 +1,544 @@
+    """
+    # Check if 'list' method support search(look for search parameter):
+    if 'search' in inspect.getargspec(service.list)[0]:
+        res = service.list(
+            search=' and '.join('{}={}'.format(k, v) for k, v in kwargs.items())
+        )
+    else:
+        res = [
+            e for e in service.list() if len([
+                 k for k, v in kwargs.items() if getattr(e, k, None) == v
+            ]) == len(kwargs)
+        ]
+
+    res = res or [None]
+    return res[0]
+
+
+def search_by_name(service, name, **kwargs):
+    """
+    Search for the entity by its name. Nested entities don't support search
+    via REST, so in case using search for nested entity we return all entities
+    and filter them by name.
+
+    :param service: service of the entity
+    :param name: name of the entity
+    :return: Entity object returned by Python SDK
+    """
+    # Check if 'list' method support search(look for search parameter):
+    if 'search' in inspect.getargspec(service.list)[0]:
+        res = service.list(
+            search="name={name}".format(name=name)
+        )
+    else:
+        res = [e for e in service.list() if e.name == name]
+
+    if kwargs:
+        res = [
+            e for e in service.list() if len([
+                k for k, v in kwargs.items() if getattr(e, k, None) == v
+            ]) == len(kwargs)
+        ]
+
+    res = res or [None]
+    return res[0]
+
+
+def wait(service, condition, fail_condition=lambda e: False, timeout=180, wait=True):
+    """
+    Wait until entity fulfill expected condition.
+
+    :param service: service of the entity
+    :param condition: condition to be fulfilled
+    :param fail_condition: if this condition is true, raise Exception
+    :param timeout: max time to wait in seconds
+    :param wait: if True wait for condition, if False don't wait
+    """
+    # Wait until the desired state of the entity:
+    if wait:
+        start = time.time()
+        while time.time() < start + timeout:
+            # Exit if the condition of entity is valid:
+            entity = service.get()
+            if condition(entity):
+                return
+            elif fail_condition(entity):
+                raise Exception("Error while waiting on result state of the entity.")
+            else:
+                time.sleep(float(wait))
+
+
+def ovirt_full_argument_spec(**kwargs):
+    """
+    Extend parameters of module with parameters which are common to all oVirt modules.
+
+    :param kwargs: kwargs to be extended
+    :return: extended dictionary with common parameters
+    """
+    spec = dict(
+        auth=dict(required=True, type='dict'),
+        timeout=dict(default=180, type='int'),
+        wait=dict(default=True, type='bool'),
+    )
+    spec.update(kwargs)
+    return spec
+
+
+def check_params(module):
+    """
+    Most modules must have either `name` or `id` specified.
+    """
+    if module.params['name'] is None and module.params['id'] is None:
```

Comment:
Just in case it's probably a good idea to use `.get()` instead of dict access syntax in case the key is unset.